### PR TITLE
Add batch.{prefix,suffix,separator} configuration properties

### PR DESCRIPTION
**Pull request by jackowitz**
These properties are similar to those in the Confluent connector
and are useful for supporting some additionally flexibility around
how exactly batch requests are formatted, including a JSON array
format when individual records are serialized as JSON.

The default values of "\n" for suffix and separator are chosen for
backwards compatibility with the current behavior when the new
properties are not specified, as confirmed by the unmodified
testBatching() integration test.

We've been running this in production without issue, specifically
with `[`, `]`, and `,` for a JSON array style batching.

### CODE REVIEW

File: src/main/java/io/aiven/kafka/connect/http/config/HttpSinkConfig.java
Comment (jackowitz) [inline src/main/java/io/aiven/kafka/connect/http/config/HttpSinkConfig.java:563]: I originally had this as the empty string like with prefix but changed to the newline since the current batching configuration does include a trailing newline after the last record. This may not be the most intuitive for new usages but provides backwards compatibility for existing connectors, so I'll defer to maintainers on which is preferred here.

Comment (tvainika) [reply src/main/java/io/aiven/kafka/connect/http/config/HttpSinkConfig.java:563]: One shouldn't need `originalsStrings` here as `getString` can return `null`. Also calling `getString` instead `originalsString` marks configuration parameter as "used". Thus something like this:
```suggestion
        String result = getString(BATCH_SUFFIX_CONFIG);
        return result != null ? result : "\n";
```

**Review by tvainika — CHANGES_REQUESTED** [review]
Thank you for the enhancements. I looked your changes, and suggesting some improvements

File: docs/sink-connector-config-options.rst
Comment (jackowitz) [reply docs/sink-connector-config-options.rst:136]: Thanks, @tvainika . I agree, the actual defaults are more meaningful. However, this documentation change was auto-generated from the `ConfigDef`s as part of the build process so I'm not sure how to override without future builds always generating a diff trying to bring them into sync. Any suggestions?

File: src/main/java/io/aiven/kafka/connect/http/config/HttpSinkConfig.java
Comment (jackowitz) [reply src/main/java/io/aiven/kafka/connect/http/config/HttpSinkConfig.java:563]: It's not so much about the `null` but rather about totally valid delimiters like `\n` or `\t` getting `trim()`ed to the empty string. For example, if the user specifies `"batch.separator": "\n"` (or `\t` anything else subject to `trim()`), then `batchSeparator()` will return the empty string, breaking batching, since the parsing logic always calls `trim()` and `getString()` works only with the trimmed values. Fortunately, the parsing logic does also stash the original string values, and the `originalsStrings()` hack was something I [found](https://github.com/confluentinc/kafka-connect-storage-cloud/blob/83c8b47f2732f4206760ee132e7e474547c990ea/kafka-connect-s3/src/main/java/io/confluent/connect/s3/S3SinkConnectorConfig.java#L786-L793) in the S3 connector from Confluent while looking at how other connectors worked around this limitation.

Comment (tvainika) [reply src/main/java/io/aiven/kafka/connect/http/config/HttpSinkConfig.java]: You're right here, I didn't see fully how this trimming was used.

File: docs/sink-connector-config-options.rst
Comment (tvainika) [reply docs/sink-connector-config-options.rst]: I didn't remember there was autogenerator here. It is better to leave as is.

**Review by tvainika — APPROVED** [review]

**Pull request merged**
